chia/types/blockchain_format/program.py

```diff
@@ -136,11 +136,37 @@ def curry(self, *args) -> "Program":
             fixed_args = [4, (1, arg), fixed_args]
         return Program.to([2, (1, self), fixed_args])
 
-    def uncurry(self) -> Tuple["Program", "Program"]:
-        r = uncurry(self)
-        if r is None:
+    def uncurry(self) -> Tuple[Program, Program]:
+        def match(o: SExp, expected: bytes) -> None:
```

Comment:
`assert_match` maybe?

Reply:
I was considering using actual `assert` (and catch it), but it would be a bit of an abuse. Perhaps it's me by "assert" implies that failure is a bug (like in systems languages).